DuckDuckGo/AppDependencyProvider.swift

```diff
@@ -72,9 +78,91 @@ class AppDependencyProvider: DependencyProvider {
+    }
+    let vpnFeatureVisibility: DefaultNetworkProtectionVisibility
+    let networkProtectionKeychainTokenStore: NetworkProtectionKeychainTokenStore
+    let networkProtectionAccessController: NetworkProtectionAccessController
+    let networkProtectionTunnelController: NetworkProtectionTunnelController
+
+    let subscriptionAppGroup = Bundle.main.appGroup(bundle: .subs)
+    
+    let connectionObserver: ConnectionStatusObserver = ConnectionStatusObserverThroughSession()
+    let vpnSettings = VPNSettings(defaults: .networkProtectionGroupDefaults)
+
+    // swiftlint:disable:next function_body_length
+    init() {
+        featureFlagger = DefaultFeatureFlagger(internalUserDecider: internalUserDecider,
+                                               privacyConfigManager: ContentBlocking.shared.privacyConfigurationManager)
+
+        // MARK: - Configure Subscription
+        let subscriptionUserDefaults = UserDefaults(suiteName: subscriptionAppGroup)!
+        let subscriptionEnvironment = SubscriptionManager.getSavedOrDefaultEnvironment(userDefaults: subscriptionUserDefaults)
+        vpnSettings.alignTo(subscriptionEnvironment: subscriptionEnvironment)
+
+        let entitlementsCache = UserDefaultsCache<[Entitlement]>(userDefaults: subscriptionUserDefaults,
+                                                                 key: UserDefaultsCacheKey.subscriptionEntitlements,
+                                                                 settings: UserDefaultsCacheSettings(defaultExpirationInterval: .minutes(20)))
+        let accessTokenStorage = SubscriptionTokenKeychainStorage(keychainType: .dataProtection(.named(subscriptionAppGroup)))
+        let subscriptionService = SubscriptionService(currentServiceEnvironment: subscriptionEnvironment.serviceEnvironment)
+        let authService = AuthService(currentServiceEnvironment: subscriptionEnvironment.serviceEnvironment)
+        let accountManager = AccountManager(accessTokenStorage: accessTokenStorage,
+                                            entitlementsCache: entitlementsCache,
+                                            subscriptionService: subscriptionService,
+                                            authService: authService)
+        if #available(iOS 15.0, *) {
+            subscriptionManager = SubscriptionManager(storePurchaseManager: StorePurchaseManager(),
+                                                      accountManager: accountManager,
+                                                      subscriptionService: subscriptionService,
+                                                      authService: authService,
+                                                      subscriptionEnvironment: subscriptionEnvironment)
+        } else {
+            // This is used just for iOS <15, it's a sort of mocked environment that will not be used.
+            subscriptionManager = SubscriptionManageriOS14(accountManager: accountManager)
+        }
```

Comment:
This is the only place where SubscriptionManager is initialised in the app, I can easily do an extension with a simple init but what's the benefit?

Reply:
I think the benefit is for hiding the complexity of what the `SubscriptionManager` is composed that is not required from the client's pov. I am in favour of minimal API of passing what is required to the initialiser and have it internally create and wire together the specific entities.
I get your point though, I leave the decision up to you.

Reply:
So.... I tried, and it was more difficult than I expected.
The accountManager is used in the accessTokenProvider closure below and that closure can't use `self.` because it is before the object init end. So the accountManager needs to be created in the current scope :/
Additionally, I have 3 iVars used in other parts of the AppDependencyProvider inits that I need to pass back and forward the SubscriptionManager init.
I'll leave this as it is for simplicity sake.